python/pyproject.toml

```diff
@@ -36,6 +36,7 @@ dependencies = [
 docs = ["markdown", "pyyaml", "sphinx", "sphinx_rtd_theme"]
 lint = ["ruff"]
 optional = ["numba"]
+petsc4py = ["petsc4py"]
```

Comment:
Maybe we could just call this `petsc = ["petsc4py"]` for short?

Reply:
`petsc` and `petssc4py` are different things.